bin/spiced/src/lib.rs

```diff
@@ -99,65 +77,9 @@ pub async fn run(args: Args) -> Result<()> {
     let mut df = runtime::datafusion::DataFusion::new();
 
     for ds in &app.datasets {
-        if ds.acceleration.is_none() && !ds.is_view() {
-            tracing::warn!("No acceleration specified for dataset: {}", ds.name);
-            continue;
-        };
-
-        let source = ds.source();
-        let source = source.as_str();
-        let params = Arc::new(ds.params.clone());
-        let data_connector: Option<Box<dyn DataConnector>> = match source {
-            "spice.ai" => Some(Box::new(
-                dataconnector::spiceai::SpiceAI::new(auth.get(source), params)
-                    .await
-                    .context(UnableToInitializeDataConnectorSnafu {
-                        data_connector: source,
-                    })?,
-            )),
-            "dremio" => Some(Box::new(
-                dataconnector::dremio::Dremio::new(auth.get(source), params)
-                    .await
-                    .context(UnableToInitializeDataConnectorSnafu {
-                        data_connector: source,
-                    })?,
-            )),
-            "localhost" | "" => None,
-            "debug" => Some(Box::new(dataconnector::debug::DebugSource {})),
-            _ => UnknownDataConnectorSnafu {
-                data_connector: source,
-            }
-            .fail()?,
-        };
-
-        let view_sql = ds.view_sql().context(InvalidSQLViewSnafu)?;
-
-        match data_connector {
-            Some(data_connector) => {
-                let data_connector = Box::leak(data_connector);
-                let data_backend = df.new_backend(ds).context(UnableToCreateBackendSnafu)?;
-
-                df.attach(ds, data_connector, data_backend).context(
-                    UnableToAttachDataConnectorSnafu {
-                        data_connector: source,
-                    },
-                )?;
-            }
-            None => {
-                if view_sql.is_some() {
-                    df.attach_view(ds).context(UnableToAttachViewSnafu)?;
-                } else {
-                    let data_backend = df.new_backend(ds).context(UnableToCreateBackendSnafu)?;
-                    df.attach_backend(&ds.name, data_backend).context(
-                        UnableToAttachDataConnectorSnafu {
-                            data_connector: source,
-                        },
-                    )?;
-                }
-            }
-        }
-
-        tracing::info!("Loaded dataset: {}", ds.name);
+        Runtime::load_dataset(ds, &auth, &mut df)
```

Comment:
Looks like you have a conflict to resolve, I think this code was changed recently.

Reply:
@mitchdevenport - merged the latest from `rust`, thanks!